src/Cabinet/Cabinet.cs

```diff
@@ -0,0 +1,207 @@
+        /// </param>
+        /// <param name="pfnseek">
+        /// Pointer to an application-defined callback function to move a file pointer to the specified location. The function should be declared using the FNSEEK macro.
+        /// </param>
+        /// <param name="cpuType">
+        /// In the 16-bit version of FDI, specifies the CPU type and can be any of the following values.
+        /// </param>
+        /// <param name="perf">
+        /// Pointer to an ERF structure that receives the error information.
+        /// </param>
+        /// <returns>
+        /// If the function succeeds, it returns a non-<see cref="IntPtr.Zero"/> HFDI context pointer; otherwise, it returns <see cref="IntPtr.Zero"/>.
+        /// </returns>
+        [DllImport(nameof(Cabinet), CallingConvention = CallingConvention.Cdecl)]
+        public static unsafe extern FdiHandle FDICreate(FNALLOC pfnalloc, FNFREE pfnfree, FNOPEN pfnopen, FNREAD pfnread, FNWRITE pfnwrite, FNCLOSE pfnclose, FNSEEK pfnseek, CpuType cpuType, ERF* perf);
+
+        /// <summary>
+        /// The FDICopy function extracts files from cabinets.
+        /// </summary>
+        /// <param name="hfdi">
+        /// A valid FDI context handle returned by the FDICreate function.
+        /// </param>
+        /// <param name="pszCabinet">
+        /// The name of the cabinet file, excluding any path information, from which to extract files.
+        /// If a file is split over multiple cabinets, FDICopy allows for subsequent cabinets to be opened.
+        /// </param>
+        /// <param name="pszCabPath">
+        /// The pathname of the cabinet file, but not including the name of the file itself. For example, "C:\MyCabs".
+        /// The contents of pszCabinet are appended to pszCabPath to create the full pathname of the cabinet.
+        /// </param>
+        /// <param name="flags">
+        /// No flags are currently defined and this parameter should be set to zero.
+        /// </param>
+        /// <param name="pfnfdin">
+        /// An application-defined callback notification function to update the application on the status of the decoder.
+        /// </param>
+        /// <param name="pfnfdid">
+        /// Not currently used by FDI. This parameter should be set to <see cref="IntPtr.Zero"/>.
+        /// </param>
+        /// <param name="pvUser">
+        /// Pointer to an application-specified value to pass to the notification function.
+        /// </param>
+        /// <returns>
+        /// If the function succeeds, it returns <see langword="true"/>; otherwise, <see langword="false"/>.
+        /// </returns>
+        [DllImport(nameof(Cabinet), CallingConvention = CallingConvention.Cdecl)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool FDICopy(FdiHandle hfdi, string pszCabinet, string pszCabPath, int flags, PFNNOTIFY pfnfdin, IntPtr pfnfdid, IntPtr pvUser);
```

Comment:
Please use pointers rather than `IntPtr`, as the `IntPtr` versions will be auto-generated.